auth0/auth0.go

```diff
@@ -32,6 +42,29 @@ func InitClientCredentials(issuer string, options ...func(*config) error) {
 	}
 
 	oauth.InitClientCredentials(issuer+"oauth/token",
+		oauth.Scopes(c.scopes...),
+		oauth.Extra(append([]string{"audience"}, c.extra...)...),
+		oauth.GetParams(func(profile map[string]string) url.Values {
+			return url.Values{
+				"audience": {profile["audience"]},
+			}
+		}))
+}
+
+// InitAuthCode sets up the Auth0 authorization code flow. Must be
+// called *after* you have called `cli.Init()`. Pass in profile-related extra
+// variables to store them alongside the default profile information.
+func InitAuthCode(clientID string, issuer string, options ...func(*config) error) {
+	var c config
+
+	for _, option := range options {
+		if err := option(&c); err != nil {
```

Comment:
I'm having trouble understanding what's going on in this and similar loops. Again go knowledge 😅 . Is this setting something on the `c` config object?

Reply:
https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis is a great deep dive on how and why functional options create a nice interface. It's one of a few different approaches and I was using the auth module to try it out. So far I like how it works.